frontend/src/hooks/useGetMetricsStepsEffect.ts

```diff
@@ -48,6 +49,14 @@ export const useGetMetricsStepsEffect = (): useGetMetricsStepsEffectInterface =>
     );
     const hasRejected = allStepsRes.some((stepInfo) => stepInfo.status === 'rejected');
     const hasFulfilled = allStepsRes.some((stepInfo) => stepInfo.status === 'fulfilled');
+    const rejectedIndices = allStepsRes.reduce((indices: number[], stepInfo, index) => {
+      if (stepInfo.status === 'rejected') {
```

Comment:
refactor `rejected` as a fixture, also refactor it in the `useGenerateReportEffect.ts`